File: 백준/Gold/1806. 부분합/부분합.java

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.*;

public class Main {

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());

        int N = Integer.parseInt(st.nextToken());
        int S = Integer.parseInt(st.nextToken());

        int[] arr = new int[N + 1];

        st = new StringTokenizer(br.readLine());

        for (int i = 0; i < N; i++) {
            arr[i] = Integer.parseInt(st.nextToken());
        }

        int lt = 0, rt = 0;
        int sum = 0;
        int answer = Integer.MAX_VALUE;

        while (lt <= rt && rt <= N) {
            if(sum < S) {
                sum += arr[rt++];
            } else {
                answer = Math.min(answer, rt - lt);
                sum -= arr[lt++];
            }
        }

        System.out.println(answer == Integer.MAX_VALUE ? 0 : answer);
    }
}

File: 백준/Gold/1806. 부분합/부분합.py
import sys

n, m = map(int, input().split())

arr = [int(s) for s in input().split()]



total = 0
answer = sys.maxsize
sum = 0
lt, rt = 0, 0

while lt <= rt and rt < n:
    sum += arr[rt]
    rt += 1

    while sum >= m:
        answer = min(answer, rt - lt)
        sum -= arr[lt]
        lt += 1

print(answer if answer != sys.maxsize else 0)